module6/hw_ig.py

```diff
@@ -0,0 +1,50 @@
+# 1. Implement a function that flatten incoming data:
+# non-iterables and elements from iterables (any nesting depth should be supported)
+# function should return an iterator (generator function)
+# don't use third-party libraries
+
+def merge_elems(*args):
+    for arg in args:
+        if hasattr(arg, '__iter__') and not isinstance(arg, str):
+            yield from merge_elems(*arg)
+        else:
+            yield arg
+
+# example input
+a = [1, 2, 3]
+b = 6
+c = 'zhaba'
+d = [[1, 2], [3, 4]]
+
+for _ in merge_elems(a, b, c, d):
```

Comment:
strings of length >1 should be processed as other iterable objects
actual output: 1 2 3 6 zhaba 1 2 3 4
expected: 1 2 3 6 z h a b a 1 2 3 4